app/controllers/events_controller.rb

```diff
@@ -25,6 +25,7 @@ def search
 
   def show
     @event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+    raise_not_found if @event.status_id == GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
```

Comment:
This reads nicer the way you have it in the internal controller with `pending_event_status_id`